mssmt/tree_test.go

```diff
@@ -822,6 +822,103 @@ func TestBIPTestVectors(t *testing.T) {
 	}
 }
 
+// TestTreeCopy tests the Copy method for both FullTree and CompactedTree,
+// including copying between different tree types.
+func TestTreeCopy(t *testing.T) {
```

Comment:
All these test cases assume that the target tree is empty. We might want to assert that in the copy routine or add test coverage for non-empty.

Reply:
Expanded to include a non-empty target tree.